src/main/resources/META-INF/resources/primefaces/datepicker/1-datepicker.js

```diff
@@ -302,12 +302,19 @@ PrimeFaces.widget.DatePicker = PrimeFaces.widget.BaseWidget.extend({
                     widget: _self,
                     handle: function(content) {
                         var dateMetadata = JSON.parse(content).dateMetadata;
+                        var pdp = _self.jq.data().primeDatePicker;
                         var disabledDates = [];
+                        var dateStyleClasses = {};
                         for (date in dateMetadata) {
+                            var parsedDate = pdp.parseOptionValue(date);
                             if (dateMetadata[date].disabled) {
-                                disabledDates.push(date);
+                                disabledDates.push(parsedDate);
+                            }
+                            if (dateMetadata[date].styleClass) {
+                              dateStyleClasses[pdp.toISO(parsedDate)] = dateMetadata[date].styleClass;
```

Comment:
Check your spacing here it looks like this line needs 2 more space indent.

Reply:
I already fixed my setting this weekend. Will fix this one.